### ISSUE TITLE
SetRegularKey: if FormattedSettings is empty {} then the transaction type defaults to AccountSet

### ISSUE THREAD

**Issue by florent-uzio** [ripple/ripple-lib#1439]
According to the documentation, the `RegularKey` field can be omitted to remove a regular key from an account.

```
RegularKey:
(Optional) A base-58-encoded Address that indicates the regular key pair to be assigned to the account. If omitted, removes any existing regular key pair from the account. Must not match the master key pair for the address.
```

https://xrpl.org/setregularkey.html#setregularkey-fields

But if we submit a transaction with an empty `FormattedSettings` object, the transaction type defaults to `AccountSet`.

The alternative is to use `prepareTransaction` instead of `prepareSettings`.

**Comment by intelliot** [comment]
@florent-uzio What if you set RegularKey to `null`?

I think it's fine to use prepareTransaction. That would be my preferred approach; we could even consider deprecating prepareSettings, as it's just a convenience wrapper over prepareTransaction.

**Comment by florent-uzio** [comment]
@intelliot I haven't tried setting it to `null` actually.
`prepareTransaction` works fine.

`prepareSettings` is convenient for some other fields such as `Domain`. I don't think `prepareTransaction` encodes the domain.
If we go with `prepareTransaction` only, we need to add more tutorials to the documentation.

**Comment by mvadari** [comment]
Resolved by #1605, which removes all `prepare` methods, including `prepareSettings`.
